### PR TITLE
Addressing multiple PCD overlay feedback

### PR DESCRIPTION
**Pull request by benjaminpkane**
Resolves comments 1, 3, 4, and 5 from https://github.com/voxel51/fiftyone/pull/2912#issuecomment-1536435096

### CODE REVIEW

**Review by brimoor — CHANGES_REQUESTED** [review]
I see the code changes you made here on github, but when I build + test locally, I'm not seeing the changes for some reason... is it user error?
Screen Shot 2023-05-05 at 4 30 27 PM

Also now the `sample tags` entry in the modal sidebar is completely missing:
Screen Shot 2023-05-05 at 4 30 40 PM

Also, unrelated but just reporting as I see things: when testing with this dataset:

```py
import fiftyone as fo
import fiftyone.zoo as foz

dataset = fo.Dataset()
qsg = foz.load_zoo_dataset("quickstart-groups", max_samples=12)

qsg_pngs = qsg.select_group_slices("left").clone()
qsg_pcds = qsg.select_group_slices("pcd").clone()

ground_truth_label = qsg_pcds.first()["ground_truth"]

g1 = fo.Group()
g1s1 = fo.Sample(filepath=qsg_pcds.first().filepath, group=g1.element("pcd1"))
g1s1["detections"] = ground_truth_label
g1s2 = fo.Sample(filepath=qsg_pcds.skip(1).first().filepath, group=g1.element("pcd2"))
g1s3 = fo.Sample(filepath=qsg_pngs.first().filepath, group=g1.element("png"))

g2 = fo.Group()
g2s1 = fo.Sample(filepath=qsg_pcds.skip(2).first().filepath, group=g2.element("pcd1"))
g2s1["detections"] = ground_truth_label
g2s2 = fo.Sample(filepath=qsg_pcds.skip(3).first().filepath, group=g2.element("pcd2"))
g2s3 = fo.Sample(filepath=qsg_pngs.skip(1).first().filepath, group=g2.element("png"))

samples = [g1s1, g1s2, g1s3, g2s1, g2s2, g2s3]

dataset.add_samples(samples)
dataset.default_group_slice = "png"

session = fo.launch_app(dataset)
```

I get the following error when using the tagging menu above the grid to add some tags *with one or more samples selected*:

```py
Traceback (most recent call last):
  File "/Users/Brian/dev/fiftyone/fiftyone/server/decorators.py", line 34, in wrapper
    response = await func(endpoint, request, data, *args)
  File "/Users/Brian/dev/fiftyone/fiftyone/server/routes/tagging.py", line 64, in post
    tags, items = await view._async_aggregate(
  File "/Users/Brian/dev/fiftyone/fiftyone/core/collections.py", line 8827, in _async_aggregate
    compiled_facet_aggs, facet_pipelines = self._build_facets(
  File "/Users/Brian/dev/fiftyone/fiftyone/core/collections.py", line 8929, in _build_facets
    field = self.get_field(root)
  File "/Users/Brian/dev/fiftyone/fiftyone/core/collections.py", line 1025, in get_field
    _, field = self._parse_field(
  File "/Users/Brian/dev/fiftyone/fiftyone/core/collections.py", line 1058, in _parse_field
    schema = self.get_field_schema(include_private=include_private)
  File "/Users/Brian/dev/fiftyone/fiftyone/core/view.py", line 692, in get_field_schema
    schema = self._get_filtered_schema(schema)
  File "/Users/Brian/dev/fiftyone/fiftyone/core/view.py", line 1433, in _get_filtered_schema
    selected_fields, excluded_fields = self._get_selected_excluded_fields(
  File "/Users/Brian/dev/fiftyone/fiftyone/core/view.py", line 1468, in _get_selected_excluded_fields
    _view = _view._add_view_stage(stage, validate=False)
  File "/Users/Brian/dev/fiftyone/fiftyone/core/view.py", line 1412, in _add_view_stage
    media_type = stage.get_media_type(self)
  File "/Users/Brian/dev/fiftyone/fiftyone/core/stages.py", line 3998, in get_media_type
    media_types = set(group_media_types.values())
AttributeError: 'NoneType' object has no attribute 'values'
```

I guess this means that the App is trying to append a `SelectGroupSlices` stage to a view that's not grouped

**Comment by brimoor** [comment]
@benjaminpkane I merged `develop` into this branch and now I see the following error when clicking on the tags action on any dataset (eg an image dataset).

I suspect this isn't related to this PR, but I'm just using this PR as a place to report it 😅 

```
Network Error

Code
500

Route
http://localhost:5151/tagging

Payload
{
  "dataset": "malaria-cell-images",
  "filters": {},
  "modal": false,
  "view": [],
  "label_fields": [
    "ground_truth",
    "predictions"
  ],
  "target_labels": false,
  "slices": [
    null
  ],
  "group_id": null,
  "sample_ids": null,
  "labels": null,
  "hidden_labels": null
}

Trace
Traceback (most recent call last):
  File "/Users/Brian/dev/fiftyone/fiftyone/server/decorators.py", line 34, in wrapper
    response = await func(endpoint, request, data, *args)
  File "/Users/Brian/dev/fiftyone/fiftyone/server/routes/tagging.py", line 38, in post
    view = fost.get_tag_view(
  File "/Users/Brian/dev/fiftyone/fiftyone/server/tags.py", line 30, in get_tag_view
    view = fosv.get_view(
  File "/Users/Brian/dev/fiftyone/fiftyone/server/view.py", line 120, in get_view
    view = view.select_group_slices(sample_filter.group.slices)
  File "/Users/Brian/dev/fiftyone/fiftyone/core/collections.py", line 5750, in select_group_slices
    return self._add_view_stage(
  File "/Users/Brian/dev/fiftyone/fiftyone/core/view.py", line 1404, in _add_view_stage
    stage.validate(self)
  File "/Users/Brian/dev/fiftyone/fiftyone/core/stages.py", line 4291, in validate
    raise ValueError("%s has no groups" % type(sample_collection))
ValueError: <class 'fiftyone.core.view.DatasetView'> has no groups
```

File: app/packages/looker-3d/src/action-bar/SliceSelector.tsx [hunk 1]
```diff
@@ -43,7 +43,7 @@ export const SliceSelector = () => {
 
   return (
     <>
-      <ActionItem title="Select pcds">
+      <ActionItem title="Select point clouds">
```
[the remark this answers is not in the document]
Comment (sashankaryal) [reply app/packages/looker-3d/src/action-bar/SliceSelector.tsx:46]: I'll take care of it and comment (2) in the subsequent commit

**Comment by benjaminpkane** [comment]
Regarding
```
Traceback (most recent call last):
  File "/Users/Brian/dev/fiftyone/fiftyone/server/decorators.py", line 34, in wrapper
    response = await func(endpoint, request, data, *args)
  File "/Users/Brian/dev/fiftyone/fiftyone/server/routes/tagging.py", line 64, in post
    tags, items = await view._async_aggregate(
  File "/Users/Brian/dev/fiftyone/fiftyone/core/collections.py", line 8827, in _async_aggregate
    compiled_facet_aggs, facet_pipelines = self._build_facets(
  File "/Users/Brian/dev/fiftyone/fiftyone/core/collections.py", line 8929, in _build_facets
    field = self.get_field(root)
  File "/Users/Brian/dev/fiftyone/fiftyone/core/collections.py", line 1025, in get_field
    _, field = self._parse_field(
  File "/Users/Brian/dev/fiftyone/fiftyone/core/collections.py", line 1058, in _parse_field
    schema = self.get_field_schema(include_private=include_private)
  File "/Users/Brian/dev/fiftyone/fiftyone/core/view.py", line 692, in get_field_schema
    schema = self._get_filtered_schema(schema)
  File "/Users/Brian/dev/fiftyone/fiftyone/core/view.py", line 1433, in _get_filtered_schema
    selected_fields, excluded_fields = self._get_selected_excluded_fields(
  File "/Users/Brian/dev/fiftyone/fiftyone/core/view.py", line 1468, in _get_selected_excluded_fields
    _view = _view._add_view_stage(stage, validate=False)
  File "/Users/Brian/dev/fiftyone/fiftyone/core/view.py", line 1412, in _add_view_stage
    media_type = stage.get_media_type(self)
  File "/Users/Brian/dev/fiftyone/fiftyone/core/stages.py", line 3998, in get_media_type
    media_types = set(group_media_types.values())
AttributeError: 'NoneType' object has no attribute 'values'
```


This has something to do with a media type is computed on views. Perhaps it has changed recently
```py
import fiftyone.core.view as fov
import fiftyone.zoo as foz

dataset = foz.load_zoo_dataset("quickstart-groups")
view = dataset.select_group_slices("png")
assert view.media_type == "image" # is image, correct
view = fov.make_optimized_select_view(view, [dataset.first().id])
assert view.media_type == "image" # should be image, is group
```

**Comment by sashankaryal** [comment]
Re, comment (2):
> Should we instead remember the slices that are currently being shown, for consistency with how everything else (I believe) is persisted?

I attempted this but this is not as easy to persist, unfortunately. I'll wait until `useStateUpdate()` refactoring is done; I believe @benjaminpkane is working on it. This has to do with how app state gets reset on page refresh or dataset changes.

**Review by sashankaryal — APPROVED** [review]
lgtm, this pr also fixes the aggregation regressions

**Comment by brimoor** [comment]
@benjaminpkane thanks for tracking down the tagging issue. I updated `make_optimized_select_view()` in such a way that the issue is resolved. FWIW I don't think this was caused by a backend change 🤷

**Review by brimoor — APPROVED** [review]
LGTM

**Pull request merged**